### PR TITLE
Updates dev setup docs to account for common pitfalls.

### PR DESCRIPTION
**Pull request by krashanoff**
There were a handful of common pitfalls with our existing dev setup docs. This PR should (hopefully) prevent these pitfalls in the future:
* Improved navigating the shell.
* **Highly recommends** using RVM, especially on WSL, since system ruby complaints were a common shared pain point.
* Accounts for minor differences between OSX, Windows, Linux.

Outside the scope of this PR, but within the scope of improving our docs: another remedy I'll be pursuing is an official video tutorial on setting up your WSL environment. I am currently sitting on the raw footage, and will try to edit it together by the end of the week.

Any and all feedback on what to add to the docs is appreciated. Will be rolling out updates as more info comes along.

### CODE REVIEW

File: _docs/dev-setup.md [hunk 1]
```diff
@@ -12,13 +12,14 @@ You may ask: why is the website so complicated to set up! You may have a point,
 
 ## Table of Contents
 
-Oh wow, this looks like a lot! But trust us, this only takes 15 minutes, max.
```
Comment (malsf21) [inline _docs/dev-setup.md:15]: Sad

File: _docs/dev-setup.md [hunk 2]
```diff
@@ -28,13 +29,53 @@ Oh wow, this looks like a lot! But trust us, this only takes 15 minutes, max.
   * [Make your Changes](#make-your-changes)
   * [Add, Commit, and Push](#add-commit-and-push)
 * [Next Steps](#next-steps)
-* [Appendix: GitHub Desktop](#appendix-github-desktop)
-* [Appendix: Installing Ruby with RVM](#appendix-installing-ruby-with-rvm)
+* [Appendix](#appendix)
```
Comment (malsf21) [inline _docs/dev-setup.md:32]: There is no appendix header, so you should add one I think!

**Review by malsf21 — APPROVED** [review]
This is just docs, so I'll give you the general LGTM; left a quick pointer on TOC (I generate mine with `gh-md-toc`) but it's not a huge blocker.

File: _docs/dev-setup.md (hunk 1 above)
Comment (krashanoff) [reply _docs/dev-setup.md:15]: One way or another, this is gonna take people starting from scratch significantly more than 15 minutes (´。＿。｀)

File: _docs/dev-setup.md (hunk 2 above)
Comment (krashanoff) [reply _docs/dev-setup.md:32]: There is, but it has no content beneath it - just used to mark the section.

**Pull request merged**